pkg/operator/controller/status.go

```diff
@@ -89,102 +87,90 @@ func (r *reconciler) syncOperatorStatus() error {
 
 // getOperatorState gets and returns the resources necessary to compute the
 // operator's current state.
-func (r *reconciler) getOperatorState() (*corev1.Namespace, []operatorv1.IngressController, []appsv1.Deployment, error) {
+func (r *reconciler) getOperatorState() (*corev1.Namespace, []operatorv1.IngressController, error) {
 	ns := manifests.RouterNamespace()
+
 	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: ns.Name}, ns); err != nil {
 		if errors.IsNotFound(err) {
-			return nil, nil, nil, nil
+			return nil, nil, nil
 		}
 
-		return nil, nil, nil, fmt.Errorf(
+		return nil, nil, fmt.Errorf(
 			"error getting Namespace %s: %v", ns.Name, err)
 	}
 
 	ingressList := &operatorv1.IngressControllerList{}
 	if err := r.client.List(context.TODO(), &client.ListOptions{Namespace: r.Namespace}, ingressList); err != nil {
-		return nil, nil, nil, fmt.Errorf(
-			"failed to list ClusterIngresses: %v", err)
-	}
-
-	deploymentList := &appsv1.DeploymentList{}
-	if err := r.client.List(context.TODO(), &client.ListOptions{Namespace: ns.Name}, deploymentList); err != nil {
-		return nil, nil, nil, fmt.Errorf(
-			"failed to list Deployment: %v", err)
+		return nil, nil, fmt.Errorf("failed to list IngressControllers: %v", err)
 	}
 
-	return ns, ingressList.Items, deploymentList.Items, nil
+	return ns, ingressList.Items, nil
 }
 
-// computeStatusConditions computes the operator's current state.
-func computeStatusConditions(conditions []configv1.ClusterOperatorStatusCondition, ns *corev1.Namespace, ingresses []operatorv1.IngressController, deployments []appsv1.Deployment) []configv1.ClusterOperatorStatusCondition {
+// computeOperatorStatusConditions computes the operator's current state.
+func computeOperatorStatusConditions(conditions []configv1.ClusterOperatorStatusCondition, ns *corev1.Namespace, ingresses []operatorv1.IngressController) []configv1.ClusterOperatorStatusCondition {
 	failingCondition := &configv1.ClusterOperatorStatusCondition{
 		Type:   configv1.OperatorFailing,
 		Status: configv1.ConditionUnknown,
 	}
 	if ns == nil {
 		failingCondition.Status = configv1.ConditionTrue
 		failingCondition.Reason = "NoNamespace"
-		failingCondition.Message = "router namespace does not exist"
+		failingCondition.Message = "ingress operator namespace does not exist"
```

Comment:
It's the operand namespace (`openshift-ingress`), not the operator namespace (`openshift-ingress-operator`).